components/profile.js

```diff
@@ -65,57 +84,401 @@ SteamCommunity.prototype.editProfile = function(settings, callback) {
 
 			switch(i) {
 				case 'name':
-					values.personaName = settings[i];
+					out[8].value = settings[i];
```

Comment:
It needs to be an array because an object can't have multiple `profile_showcase[]` keys.

Reply:
And the switch statement is there in the original too and I assume it is because if you don't iterate the properties and instead assign them f.e. `variable.city=settings.city` it would require that city is defined each time it is submitted. With the current way you can leave out city to keep the old value.